### PR TITLE
dump eth headers and libs-rs/eth-client test use dumped data

### PR DESCRIPTION
**Pull request by ailisp**
Address #58

### CODE REVIEW

File: environment/commands/dump-eth-headers/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const fs = require('fs').promises;
+
+class DumpETHHeaders {
+  static async execute({path, startBlock, endBlock, ethNodeUrl}) {
+    console.log('here')
+    let web3 = new Web3(ethNodeUrl);
+    console.log('there')
+    console.log(path)
+    console.log(endBlock)
+    for (let b = startBlock; b <= endBlock; b++) {
```
Comment (nearmax) [inline environment/commands/dump-eth-headers/index.js:12]: Please add functionality of when `endBlock` is not specified to use the most recent block.

File: environment/commands/dump-eth-headers/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const Web3 = require('web3');
+const Path = require('path');
+const fs = require('fs').promises;
+
+class DumpETHHeaders {
+  static async execute({path, startBlock, endBlock, ethNodeUrl}) {
+    console.log('here')
```
Comment (nearmax) [inline environment/commands/dump-eth-headers/index.js:7]: Please remove debugging prints.

File: environment/commands/dump-eth-headers/index.js (hunk 1 above)
Comment (ailisp) [reply environment/commands/dump-eth-headers/index.js:12]: This was not the upto date one, i'm doing some cleanup and undo changes in ethrelay (which was using the subscribe method)

**Comment by ailisp** [comment]
To test:
```
node index.js dump-eth-headers --eth-node-url wss://mainnet.infura.io/ws/v3/$INFURA_PROJECT_ID --path /tmp/data --start-block=-10

# in lib-rs
ETH_HEADER_DIR=/tmp/data cargo test --package eth-client --lib -- tests::predumped_block_can_be_added --exact --nocapture
```

File: environment/commands/dump-eth-headers/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      endBlock = Number(endBlock);
+    }
+    if (!startBlock) {
+      startBlock = Math.max(0, Number(endBlock) - 43000 + 1);
+    } else if (Number(startBlock)<0) {
+      startBlock = Math.max(0, Number(endBlock) + Number(startBlock) + 1)
+    } else {
+      startBlock = Number(startBlock);
+    }
+    console.log(`Download block ${endBlock} down to ${startBlock} to ${path}, totally ${endBlock - startBlock + 1} blocks`)
```
Comment (nearmax) [inline environment/commands/dump-eth-headers/index.js:21]:
```suggestion
    console.log(`Downloading block ${endBlock} down to ${startBlock} to ${path}. ${endBlock - startBlock + 1} blocks in total.`)
```

**Comment by ailisp** [comment]
@nearmax addressed your comment. Also I found if blocks are more than 20-30, get error of:
```
thread 'tests::predumped_block_can_be_added' panicked at 'called `Result::unwrap()` on an `Err` value: HostError(GasLimitExceeded)', /Users/bo/.cargo/registry/src/github.com-1ecc6299db9ec823/near-bindgen-0.4.2/src/environment/mocked_blockchain.rs:133:9
```
How to increase prepaid gas in this test?

**Pull request merged**